packages/datadog-instrumentations/src/ai.js

```diff
@@ -0,0 +1,124 @@
+'use strict'
+
+const { addHook } = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+const { channel, tracingChannel } = require('dc-polyfill')
+const toolCreationChannel = channel('dd-trace:vercel-ai:tool')
+
+const TRACED_FUNCTIONS = {
+  generateText: wrapWithTracer,
+  streamText: wrapWithTracer,
+  generateObject: wrapWithTracer,
+  streamObject: wrapWithTracer,
+  embed: wrapWithTracer,
+  embedMany: wrapWithTracer,
+  tool: wrapTool
+}
+
+const vercelAiTracingChannel = tracingChannel('dd-trace:vercel-ai')
+const vercelAiSpanSetAttributesChannel = channel('dd-trace:vercel-ai:span:setAttributes')
+
+const noopTracer = {
+  startActiveSpan () {
+    const fn = arguments[arguments.length - 1]
+
+    const span = {
+      end () {},
+      setAttributes () { return this },
+      addEvent () { return this },
+      recordException () { return this },
+      setStatus () { return this }
+    }
+
+    return fn(span)
+  }
+}
```

Comment:
i guess this could be extracted out into an otel noop tracer that could be shared

Reply:
i've cleaned this up, although it's still local to the instrumentation, but i'm gonna resolve for now anyways since it's only used in this instrumentation. if we need a no-op dummy otel tracer for other instrumentations down the line i think we can refactor then.

Reply:
> Why not just returning a fake span directly?
mostly because we could be patching an actual otel-compatible tracer that someone is already using. i agree if we were just concerned about a dummy tracer/spans, then yeah we could patch in-place with respect to the dummy tracer, but i did it this way bc someone could actually be using a real tracer. lmk if that answers your question!